RIOT-OS/RIOT
# cpu/stm32: cleanup timer structure in vendor headers

### Contribution description

This PR restores the timer structure in all STM32 vendor headers and adapts the timer peripheral driver accordingly.




### Testing procedure

- A green Murdock
- Run the timer related tests on STM32 boards to check nothing is broken
- Verify that the code size remains unchanged




### Issues/PRs references

None
